reference/gdscript.rst

```diff
@@ -771,32 +772,32 @@ There are 6 pattern types:
 - array pattern
     matches an array. Every single element of the array pattern is a pattern itself so you can nest them.
     
-    First the length of the array is tested. It has to be the same like the pattern, otherwise the pattern doesn't match.
+    The length of the array is tested first, it has to be the same size as the pattern, otherwise the pattern don't match.
 
     **Open-ended array**: An array can be bigger than the pattern by making the last subpattern ``..``
     
     Every subpattern has to be comma seperated. ::
     
         match x:
             []:
-                print("emtpy array")
+                print("empty array")
             [1, 3, "test", null]:
                 print("very specific array")
             [var start, _, "test"]:
-                print("first element is ", start, " ande the last is \"test\"")
+                print("first element is ", start, ", and the last is \"test\"")
             [42, ..]:
                 print("open ended array")
     
 - dictionary pattern
-    works similar to the array pattern. Every key has to be a constant pattern.
+    Works in the same was as the array pattern. Every key has to be a constant pattern.
 
-    First the size of the dictionary is tested. It has to be the same like the pattern, otherwise the pattern doesn't match.
+    The size of the dictionary is tested first, it has to be the same size as the pattern, otherwise the pattern don't match.
 
     **Open-ended dictionary**: A dictionary can be bigger than the pattern by making the last subpattern ``..``
 
     Every subpattern has to be comma seperated.
 
-    If you don't specify a value then only the existance of the key is checked (treating the dictionary like a set).
+    If you don't specify a value, then only the existance of the key is checked (treating the dictionary like a set).
```

Comment:
Well, it enables the use of dictionaries as sets, but it suggests that the whole dictionary is used as a set then. It's clear that it's per-entry without the parans